### PR TITLE
feat:History functionality

### PR DESCRIPTION
**Pull request by sharmayatendra**
# Changes

- history feature added

image

### CODE REVIEW

File: src/components/Modal/Modal.jsx [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+  const { pathname } = useLocation()
+  const {watchlaterVideos,setWatchlaterVideos} = useWatchlater()
+
+  const removeFromHistory = async() => {
+    if(token) {
+      try{
+        const resp = await axios.delete(`/api/user/history/${videoId}` , {
+          headers: {authorization: token}
+        })
+        console.log(resp);
```
Comment (tushargoyal812) [inline src/components/Modal/Modal.jsx:25]: remove console

File: src/components/VideoCard/VideoCard.jsx [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
     const [showModal,setShowModal] = useState(false);
+    const {setHistoryVideos} = useHistory()
+    const {state: {token}} = useAuth()
+
+    const addToHistory = async (video) => {
+        try{
+            const resp = await axios.post('/api/user/history' , {video} , {
+                headers: {authorization: token}
+            })
+            console.log(resp)
```
Comment (tushargoyal812) [inline src/components/VideoCard/VideoCard.jsx:19]: also here

File: src/context/watchlater-context.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    )
+}
+
+const addToWatchLater = async (video,token,setWatchlaterVideos,navigate) => {
+    if(token) {
+        try{
+            const resp = await axios.post('/api/user/watchlater' , {video} , {
+              headers: {authorization: token}
+            })
+            console.log(resp);
```
Comment (tushargoyal812) [inline src/context/watchlater-context.js:21]: also here

**Review by tushargoyal812 — COMMENTED** [review]
good to merge

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

**Pull request merged**